stonesoup/types/track.py

```diff
@@ -140,3 +155,13 @@ def _update_metadata_from_state(self, state):
                 hypothesis = state.hypothesis
                 if hypothesis and hypothesis.measurement.metadata is not None:
                     self.metadata.update(hypothesis.measurement.metadata)
+
+    @property
+    def root_tracks(self) -> set:
```

Comment:
Consider not doing this. `source_tracks` vs `root_tracks` is unnecessary overspecification. If it isn't give an example of how you are using them and why the properties need to be distinct.

My view remains that a `source_tracks = tree()` is better with `root_tracks` if you need them identified as leaf nodes.